packages/arb-token-bridge-ui/src/hooks/useDestinationAddressDenylist.ts

```diff
@@ -0,0 +1,141 @@
+import { useMemo } from 'react'
+import { l2Networks } from '@arbitrum/sdk/dist/lib/dataEntities/networks'
+
+import { useNetworksAndSigners } from './useNetworksAndSigners'
+import { useTokenLists } from './useTokenLists'
+import { useAppState } from '../state'
+
+export const useDestinationAddressDenylist = () => {
+  const {
+    app: { isDepositMode }
+  } = useAppState()
+  const {
+    l2: { network: l2Network }
+  } = useNetworksAndSigners()
+  const tokenLists = useTokenLists(l2Network.id)
+
+  const tokenListsAddresses = useMemo(() => {
+    return Array.from(
+      new Set(
+        (tokenLists.data || [])
+          .map(list => list.tokens.map(token => token.address))
+          .flat()
+      )
+    )
+  }, [tokenLists])
+
+  const result = useMemo(() => {
```

Comment:
If we only use it to know if an address is in this list, then we might want to use `Set` instead, then we can use
```
const destinationAddressDenylist = useDestinationAddressDenylist()
if (destinationAddressDenylist.has(address)) { ... }
```